pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -313,16 +353,25 @@ func AddURLsConfigMap(revised *corev1.ConfigMap, crName string, newURLs []corev1
 }
 
 //DeleteURLsConfigMap for deleting server peers to the monitoring console or scaling down
-func DeleteURLsConfigMap(revised *corev1.ConfigMap, crName string, newURLs []corev1.EnvVar, deleteCR bool) {
+func DeleteURLsConfigMap(revised *corev1.ConfigMap, crName string, newURLs []corev1.EnvVar, deleteCR bool) []string {
+	var deletedPeers []string
 	for _, url := range newURLs {
 		currentURLs := strings.Split(revised.Data[url.Name], ",")
 		sort.Strings(currentURLs)
 		for _, curr := range currentURLs {
-			//scale DOWN
-			if strings.Contains(curr, crName) && !strings.Contains(url.Value, curr) && !deleteCR {
-				revised.Data[url.Name] = strings.ReplaceAll(revised.Data[url.Name], curr, "")
-			} else if strings.Contains(curr, crName) && deleteCR {
-				revised.Data[url.Name] = strings.ReplaceAll(revised.Data[url.Name], url.Value, "")
+			if strings.Contains(curr, crName) {
+				if deleteCR {
+					revised.Data[url.Name] = strings.ReplaceAll(revised.Data[url.Name], url.Value, "")
+					if url.Name != "SPLUNK_MULTISITE_MASTER" && url.Name != "SPLUNK_DEPLOYER_URL" && url.Name != "SPLUNK_SEARCH_HEAD_CAPTAIN_URL" && url.Name != "SPLUNK_SITE" {
```

Comment:
I think the inclusion list check may give better clarity here, OR else, can we have a comment here?

Reply:
Done